Sphinx 4.0 regression: warning for `autosummary_generate` with list type

```
# conf.py
autosummary_generate = ['Python-API.rst']
```

Sphinx 4 yields the following warning
```
Warning, treated as error:
The config value `autosummary_generate' has type `list'; expected `bool'.
make: *** [Makefile:20: html] Error 2
```
However, official latest docs still say this value can has `list` type (as it was possible before 4 version).
> Can also be a list of documents for which stub pages should be generated.
    https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html#confval-autosummary_generate

--- a/CHANGES
+++ b/CHANGES
@@ -16,6 +16,9 @@ Features added
 Bugs fixed
 ----------
 
+* #9188: autosummary: warning is emitted if list value is set to
+  autosummary_generate
+
 Testing
 --------
 

--- a/sphinx/ext/autosummary/__init__.py
+++ b/sphinx/ext/autosummary/__init__.py
@@ -773,7 +773,7 @@ def setup(app: Sphinx) -> Dict[str, Any]:
     app.connect('builder-inited', process_generate_options)
     app.add_config_value('autosummary_context', {}, True)
     app.add_config_value('autosummary_filename_map', {}, 'html')
-    app.add_config_value('autosummary_generate', True, True, [bool])
+    app.add_config_value('autosummary_generate', True, True, [bool, list])
     app.add_config_value('autosummary_generate_overwrite', True, False)
     app.add_config_value('autosummary_mock_imports',
                          lambda config: config.autodoc_mock_imports, 'env')

